src/content/reference/device-os/firmware.md

```diff
@@ -405,1061 +405,1306 @@ curl https://api.particle.io/v1/devices/0123456789abcdef/brew \
+Each event delivery attempt to a subscription handler uses one Data Operation from your monthly or yearly quota. Setting up the subscription does not use a Data Operations. You should take advantage of the event prefix to avoid delivering events that you do not need. If poor connectivity results in multiple attempts, it could result in multiple Data Operations, up to 3. If the device is currently marked as offline, then no attempt will be made and no Data Operations will be incurred.
 
-- If you are using [`SYSTEM_MODE(AUTOMATIC)`](#system-modes) (the default if you do not specify), or `SEMI_AUTOMATIC` you generally do not need to `Particle.process()` unless your code blocks and prevents loop from returning and does not use `delay()` in any inner blocking loop. In other words, if you block `loop()` from returning you must call either `delay()` or `Particle.process()` within your blocking inner loop.
+If you have multiple devices that subscribe to a hook-response but only want to monitor the response to their own request, as opposed to any device in the account sharing the webhook, you should include the Device ID in the custom hook response as described [here](/reference/cloud-apis/webhooks/#responsetopic). This will assure that you will not consume Data Operations for webhooks intended for other devices.
 
-- If you are using `SYSTEM_MODE(MANUAL)` you must call `Particle.process()` frequently, preferably on any call to `loop()` as well as any locations where you are blocking within `loop()`.
+```cpp
+Particle.subscribe(System.deviceID() + "/hook-response/weather/", myHandler);
+```
 
-`Particle.process()` checks for the incoming messages from the Cloud,
-and processes any messages that have come in. It also sends keep-alive pings to the Cloud,
-so if it's not called frequently, the connection to the Cloud may be lost.
+- It is possible to receive an event more than once. The most common reason is a lost ACK, which will cause the device to send the event again. Storing a unique identifier in the event payload may help code defensively for this possibility.
+- It is possible that events will arrive out-of-order. The most common cause is retransmission, but it can also occur because events can flow through different redundant servers, each with slightly difference latency, so it's possible that two event sent rapidly will arrive out-of-order as well. This is common for multi-part webhook responses.
 
-It will also update the [ApplicationWatchdog](/reference/device-os/api/watchdog-application/watchdog-application/) timer using `ApplicationWatchdog::checkin()`.
 
-### Particle.syncTime()
+### Subscribe (with content type) - Subscribe
 
-{{api name1="Particle.syncTime"}}
+```cpp
+// PROTOTYPES
+bool subscribe(const char* name, particle::EventHandlerWithContentType handler);
+bool subscribe(const char* name, particle::EventHandlerWithContentTypeFn handler);
```

Comment:
> particle::EventHandlerWithContentType
> particle::EventHandlerWithContentTypeFn

Nit: Can we not expose these type names and just document it by example what callback signatures are expected? Definitely feel free to keep it the way it is if you feel doing so would complicate things anyhow. It's just that I would like to keep some room for renaming and moving things around as this API evolves.

Reply:
I put an example first. The full prototypes are listed farther down now.